tinygrad/nn/state.py

```diff
@@ -16,7 +16,7 @@ def safe_load_metadata(fn:Union[Tensor,str]) -> Tuple[Tensor, int, Any]:
 
 def safe_load(fn:Union[Tensor,str]) -> Dict[str, Tensor]:
   t, json_len, metadata = safe_load_metadata(fn)
-  return {k:t[8+json_len+v['data_offsets'][0]:].cast(safe_dtypes[v['dtype']])[:prod(v['shape'])].reshape(v['shape']) for k,v in metadata.items() if k != "__metadata__"}
+  return {k:t[8+json_len+v['data_offsets'][0]:].cast(safe_dtypes[v['dtype']])[:prod(v['shape'])].reshape(v['shape']).bitcast(dtypes.uint16).to("CPU").cast(dtypes.uint32).mul(1<<16).bitcast(dtypes.float32).to(Device.DEFAULT).half() if v['dtype'] == "BF16" else t[8+json_len+v['data_offsets'][0]:].cast(safe_dtypes[v['dtype']]) [:prod(v['shape'])].reshape(v['shape']) for k,v in metadata.items() if k != "__metadata__"}
```

Comment:
This needs to be abstracted out

Reply:
Thank you for the feedback, will work on it